source/DistanceMap.cpp

```diff
@@ -182,7 +184,7 @@ void DistanceMap::Init(const Ship *ship)
 		{
 			bool hasWormhole = false;
 			for(const StellarObject &object : ship->GetSystem()->Objects())
-				if(object.HasSprite() && object.HasValidPlanet() && object.GetPlanet()->IsWormhole())
+				if(object.HasSprite() && object.HasValidLandable() && isa<Wormhole>(object.GetLandable()))
```

Comment:
I feel that Wormholes are not properly separated off in this PR. Maybe the amount of casts can be reduced by adding an HasValidWormhole() to object (similar to HasValidPlanet())?

Reply:
I guess I could! Then I can also get rid of the `assert`s I believe if I introduce `HasValidWormhole` and `GetWormhole` and then the same for planets. Thanks!